compiler/src/openxla/compiler/nvgpu/Dialect/FlowTransformExtension/IR/FlowTransformExtension.cpp

```diff
@@ -0,0 +1,313 @@
+// Copyright 2023 The IREE Authors
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include "openxla/compiler/nvgpu/Dialect/FlowTransformExtension/IR/FlowTransformExtension.h"
+
+#include "iree-dialects/Transforms/TransformMatchers.h"
+#include "iree/compiler/Dialect/Flow/IR/FlowDialect.h"
+#include "iree/compiler/Dialect/Flow/IR/FlowOps.h"
+#include "llvm/Support/Debug.h"
+#include "mlir/Dialect/Linalg/IR/Linalg.h"
+#include "mlir/Dialect/Transform/IR/TransformDialect.h"
+#include "mlir/Dialect/Transform/IR/TransformInterfaces.h"
+#include "mlir/Interfaces/SideEffectInterfaces.h"
+
+#define DEBUG_TYPE "nvgpu-flow-transform-extension"
+#define DBGS() llvm::dbgs() << "[" DEBUG_TYPE "]"
+
+namespace mlir {
+namespace iree_compiler {
+namespace IREE {
+
+void transform_dialect::FilterOutAlreadyInDispatchRegionOp::getEffects(
+    SmallVectorImpl<MemoryEffects::EffectInstance> &effects) {
+  transform::onlyReadsHandle(getTarget(), effects);
+  transform::producesHandle(getTransformed(), effects);
+  transform::onlyReadsPayload(effects);
+}
+
+DiagnosedSilenceableFailure
+transform_dialect::FilterOutAlreadyInDispatchRegionOp::apply(
+    transform::TransformResults &results, transform::TransformState &state) {
+  RaggedArray<Operation *> inputs;
+  for (Value operand : getTarget()) {
+    inputs.push_back(state.getPayloadOps(operand));
+    for (Operation *payload : inputs.at(inputs.size() - 1)) {
+      if (!payload->getParentOfType<Flow::DispatchWorkgroupsOp>()) continue;
+
+      for (OpResult result : getResults()) {
+        results.set(result, {});
+      }
+      return DiagnosedSilenceableFailure::success();
+    }
+  }
+
+  for (OpResult result : getResults()) {
+    results.set(result, inputs.at(result.getResultNumber()));
+  }
+
+  return DiagnosedSilenceableFailure::success();
+}
+
+LogicalResult transform_dialect::FilterOutAlreadyInDispatchRegionOp::verify() {
+  if (getNumOperands() != getNumResults()) {
+    return emitOpError() << "expects the same number of operands and results";
+  }
+  return success();
+}
+
+///////////////////////////////////////////////////////////////////////////////
+//
+// The following code is mostly copied from IREE with a minor adaptation:
+// the matcher doesn't include `allTileableOpsCaptured` so it can run on the
```

Comment:
Let's add a 

TODO: Extend IREE API to optionally take a boolean to specify `allTileableOpsCaptured` and avoid duplication.